core/src/kura.rs

```diff
@@ -362,15 +348,30 @@ impl Unlocked {
 
     /// Try to acquire lockfile and convert self to [`Locked`]
     fn lock(self) -> Result<Locked> {
+        let path = self.join(LOCK_FILE_NAME);
         if let Err(e) = fs::File::options()
             .read(true)
             .write(true)
             .create_new(true)
-            .open(self.join(LOCK_FILE_NAME))
+            .open(path.clone())
         {
             match e.kind() {
                 std::io::ErrorKind::AlreadyExists => Err(Error::Locked(self.0)),
-                _ => Err(e.into()),
+                std::io::ErrorKind::NotFound => {
+                    std::fs::create_dir_all(self.0.clone())
+                        .map_err(|e| Error::IO(e, path.clone()))?;
```

Comment:
Probably should be `MkDir` error, no?

Reply:
No. Because we're checking for if the directory exists and then creating. 

In general I don't like the design of `std::io::error`, and would gladly change this to something more palatable

Reply:
Hmm, okay, i was concerned because in `create_files_if_they_do_not_exist` we return `MkDir` error for the same operation.

Reply:
Good point. I didn't notice that.